benches/src/main.rs

```diff
@@ -0,0 +1,235 @@
+    let cli = Cli::parse();
+
+    // Check if hyperfine is installed
+    let hyperfine_check = Command::new("hyperfine").arg("--version").output();
+    if hyperfine_check.is_err() || !hyperfine_check.unwrap().status.success() {
+        eyre::bail!(
+            "hyperfine is not installed. Please install it first: https://github.com/sharkdp/hyperfine"
+        );
+    }
+
+    // Determine versions to test
+    let versions = if let Some(v) = cli.versions {
+        v
+    } else {
+        FOUNDRY_VERSIONS.iter().map(|&s| s.to_string()).collect()
+    };
+
+    // Get repo configurations
+    let repos = if let Some(repo_specs) = cli.repos.clone() {
+        repo_specs
+            .iter()
+            .map(|spec| RepoConfig::try_from(spec.as_str()))
+            .collect::<Result<Vec<_>>>()?
+    } else {
+        BENCHMARK_REPOS.clone()
+    };
+
+    sh_println!("🚀 Foundry Benchmark Runner");
+    sh_println!("Running with versions: {}", versions.join(", "));
+    sh_println!(
+        "Running on repos: {}",
+        repos.iter().map(|r| format!("{}/{}", r.org, r.repo)).collect::<Vec<_>>().join(", ")
+    );
+
+    // Install versions if requested
+    if cli.force_install {
+        install_foundry_versions(&versions)?;
+    }
+
+    // Determine benchmarks to run
+    let benchmarks = if let Some(b) = cli.benchmarks {
+        b.into_iter().filter(|b| ALL_BENCHMARKS.contains(&b.as_str())).collect()
+    } else {
+        // Default: run all benchmarks except fuzz tests and coverage (which can be slow)
+        vec!["forge_test", "forge_build_no_cache", "forge_build_with_cache"]
+            .into_iter()
+            .map(String::from)
+            .collect::<Vec<_>>()
+    };
+
+    sh_println!("Running benchmarks: {}", benchmarks.join(", "));
+
+    let mut results = BenchmarkResults::new();
+    // Set the first version as baseline
+    if let Some(first_version) = versions.first() {
+        results.set_baseline_version(first_version.clone());
+    }
+
+    // Run benchmarks for each version
+    for version in &versions {
+        sh_println!("🔧 Switching to Foundry version: {version}");
+        switch_version_safe(version)?;
+
+        // Verify the switch and capture full version details
+        let current = get_forge_version()?;
+        sh_println!("Current version: {}", current.trim());
+
+        // Get and store the full version details with commit hash and date
+        let version_details = get_forge_version_details()?;
+        results.add_version_details(version, version_details);
+
+        // Create a list of all benchmark tasks
+        let benchmark_tasks: Vec<_> = repos
+            .iter()
+            .flat_map(|repo| {
+                benchmarks.iter().map(move |benchmark| (repo.clone(), benchmark.clone()))
+            })
+            .collect();
+
+        sh_println!("Running {} benchmark tasks in parallel...", benchmark_tasks.len());
+
+        // Run all benchmarks in parallel
```

Comment:
should we run them in parallel?

forge uses multithreading so this could interfere?